### ISSUE TITLE
Not able to save posts with nested docs plugin and drafts on postgres

### ISSUE THREAD

**Issue by maevic** [payloadcms/payload#4913]
### Link to reproduction

https://github.com/maevic/payload/tree/nested-docs-with-drafts

### Describe the Bug

Hi, when adding the nested docs plugin and enable drafts while using postgres as a database, it's not possible to save existing items or add new ones to the collection.

Here is a little video demonstrating the behavior.

https://github.com/payloadcms/payload/assets/15337940/46c68345-1f09-45e8-832f-65346b576295

### To Reproduce

```shell
git clone https://github.com/maevic/payload.git
cd payload
git switch nested-docs-with-drafts
pnpm install
docker run --rm -p 5432:5432 --env POSTGRES_USER=admin --env POSTGRES_PASSWORD=123 --env POSTGRES_DB=db --name test-postgres postgres:16.0
pnpm run dev _community
``` 

- Visit [http://localhost:3000/admin/collections/posts/1](http://localhost:3000/admin/collections/posts/1)
- There should be versions, but there aren't any
- After a click on the publish button, there are still no versions
- The posts collection page shows no items

### Payload Version

2.8.2

### Adapters and Plugins

db-postgres, plugin-nested-docs

**Comment by spencerxl** [comment]
Same issue. The page/post appears to be saved correctly in the db, but `draft=true` in the collection query
(`/api/pages?depth=0&**draft=true**&limit=10&invoke=1523c4e4-e7d1-4f79...`)

is returning 0 results, rather than returning the most recent posts with drafts.

**Comment by maevic** [comment]
I believe the tables `posts`, `_posts_v`, `_posts_v_rels` and `_posts_v_version_breadcrumbs` should have entries after saving a post. In my case only the `posts` table does.

**Comment by spencerxl** [comment]
Ah, you are 100% right, I lost track of what I'd disabled/re-enabled trying to isolate the problem. Thanks for the correction!

**Comment by denolfe** [comment]
@maevic Thanks for the submission. I believe this one was fixed by #5012 . I rebased `main` onto your recreation branch, and it appeared to function as expected (versions showing).

This fix will be in the next release.

**Comment by maevic** [comment]
Tried it and it works, thanks so much.
